### DISCUSSION TITLE
help: mac runner image doesnt specify how to set up actions runner service

### DISCUSSION THREAD

**Discussion by zoezhangmattr** [actions/virtual-environments#10926] [Image configuration]
hey , i was checking how to set up our self hosted runners, i was wondering how does GitHub download the action runners package, set it up as a service and make sure the service has access to all the bashrc environment variables setting. 
do you have any suggestions or solutions please. 
thank you

**Comment by vaishalipawar1490** [comment]
Hi @zoezhangmattr 
You can check the official GitHub documentation for setting up and managing self-hosted runners here. 
https://docs.github.com/en/actions/hosting-your-own-runners

This guide provides step-by-step instructions on how to download, configure, and manage self-hosted runners, as well as troubleshoot any issues you might encounter.

Please check if this finds useful !
